tests/conftest.py

```diff
@@ -0,0 +1,169 @@
+# Copyright (C) 2017-2022  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+
+from typing import Union
+
+import pytest
+import numpy as np
+import sklearn
+from sklearn.datasets import make_multilabel_classification
+from sklearn.model_selection import train_test_split
+from sklearn.multiclass import OneVsRestClassifier
+from sklearn.linear_model import LogisticRegression
+
+from cleanlab import count
+from cleanlab.benchmarking.noise_generation import (
+    generate_noise_matrix_from_trace,
+    generate_noisy_labels,
+)
+from cleanlab.internal.util import train_val_split
+
+
+def multilabel_py(y):
+    unique_labels = np.unique(y, axis=0)
+    n = y.shape[0]
+    multilabel_counts = [
+        np.sum(np.prod([label[i] == y[:, i] for i in range(y.shape[1])], axis=0))
+        for label in unique_labels
+    ]
+    return np.array(multilabel_counts) / float(n)
+
+
+DATASET_KWARGS = {
+    "n_samples": 500,
+    "n_features": 3,
+    "n_classes": 3,
+    "n_labels": 2,
+    "length": 500,
+    "allow_unlabeled": True,
+    "sparse": False,
+    "return_indicator": "dense",
+    "return_distributions": True,
+}
+
+
+def get_split_generator(labels, cv):
+    unique_labels = np.unique(labels, axis=0)
+    label_to_index = {tuple(label): i for i, label in enumerate(unique_labels)}
+    multilabel_ids = np.array([label_to_index[tuple(label)] for label in labels])
+    split_generator = cv.split(X=multilabel_ids, y=multilabel_ids)
+    return split_generator
+
+
+def train_fold(X, labels, *, clf, pred_probs, cv_train_idx, cv_holdout_idx):
+    clf_copy = sklearn.base.clone(clf)
+    X_train_cv, X_holdout_cv, s_train_cv, _ = train_val_split(
+        X, labels, cv_train_idx, cv_holdout_idx
+    )
+    clf_copy.fit(X_train_cv, s_train_cv)
+    pred_probs[cv_holdout_idx] = clf_copy.predict_proba(X_holdout_cv)
+
+
+def get_cross_validated_multilabel_pred_probs(X, labels, *, clf, cv):
+    split_generator = get_split_generator(labels, cv)
+    pred_probs = np.zeros(shape=labels.shape)
+    for cv_train_idx, cv_holdout_idx in split_generator:
+        train_fold(
+            X,
+            labels,
+            clf=clf,
+            pred_probs=pred_probs,
+            cv_train_idx=cv_train_idx,
+            cv_holdout_idx=cv_holdout_idx,
+        )
+    return pred_probs
```

Comment:
is this functionality similar to : 
`from sklearn.model_selection import cross_val_predict
pred_probs = cross_val_predict(clf, X_train, y_train, method="predict_proba")`

Reply:
@aditya1503  Does the `sklearn.cross_val_predict()` work well for multilabel?  I've never tried it.  For instance how does it do a stratified split in the multilabel setting?

 If it does it well, then let's just have this function internally convert the labels to proper format for `sklearn.cross_val_predict()` and then call that.

Reply:
It worked well for me, 
It gave consistent results, I need to see how it does the splitting, will run some tests
I used int2onehot(labels), 

I'll also see if sampling works as well with encoding labels to unique class names

Reply:
I'd try it with rare classes and check that it is actually stratifying the split properly

Reply:
`sklearn.model_selection.StratifiedKFold` ([link](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.StratifiedKFold.html)) does not support (binarized) multi-labels.

This is reflected in the docstring for `cv` in `cross_val_predict` ([link](https://github.com/scikit-learn/scikit-learn/blob/36958fb240fbe435673a9e3c52e769f01f36bec0/sklearn/model_selection/_validation.py#L847-L859)).

---

@aditya1503 It's not clear to me what part of the function you think `cross_val_predict` should replace.
Did you have the following in mind?

```python
def get_cross_validated_multilabel_pred_probs(X, labels, *, clf, cv):
    split_generator = get_split_generator(labels, cv)
    pred_probs = cross_val_predict(clf, X, labels, cv=split_generator, method="predict_proba") 
    return pred_probs
```